lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -200,10 +202,50 @@ getTransaction = isolate 43 $ do
             other -> fail $ "Invalid witness type: " ++ show other
 
 
+putTransaction :: (Tx, [TxWitness]) -> Put
+putTransaction (tx, witnesses) = do
+    putTokenTransfer tx
+    mapM_ putWitness witnesses
+
+putWitness :: TxWitness -> Put
+putWitness witness =
+    case witness of
+        PublicKeyWitness xPub (Hash sig) -> do
+            putWord8 1
+            putByteString xPub
+            putByteString sig
+        -- TODO: note that we are missing new address type witness:
+        --  * https://github.com/input-output-hk/rust-cardano/blob/3524cfe138a10773caa6f0effacf69e792f915df/chain-impl-mockchain/doc/format.md#witnesses
+        --  * https://github.com/input-output-hk/rust-cardano/blob/e0616f13bebd6b908320bddb1c1502dea0d3305a/chain-impl-mockchain/src/transaction/witness.rs#L23
+        ScriptWitness _ -> error "unimplemented: serialize script witness"
+        RedeemWitness _ -> error "unimplemented: serialize redeem witness"
```

Comment:
I am not sure atm how rust node covers these two cases. Maybe they are encoded under old or normal utxo witness scheme